libcudacxx/include/cuda/std/detail/libcxx/include/__type_traits/is_convertible.h

```diff
@@ -69,7 +69,10 @@ _LIBCUDACXX_INLINE_VAR constexpr bool is_convertible_v<volatile _Ty&, const vola
 
 namespace __is_convertible_imp
 {
+
+_LIBCUDACXX_NV_DIAG_SUPPRESS(3013) // a volatile function parameter is deprecated
 template <class _Tp> _LIBCUDACXX_INLINE_VISIBILITY void  __test_convert(_Tp);
+_LIBCUDACXX_NV_DIAG_DEFAULT(3013) // a volatile function parameter is deprecated
```

Comment:
This will also reset suppression on the user side.

Reply:
I had a look at the cuda documentation again and we have a winner:
```
The following pragmas may be used to save and restore the current diagnostic pragma state:
#pragma nv_diagnostic push
#pragma nv_diagnostic pop
```

Reply:
So it seems that this is not supported by earlier nvcc that only has ``#pragma diagnostic`

This also seems to not be supported by nvc++